docs/DeveloperGuide.md

```diff
@@ -487,7 +487,76 @@ The following activity diagram summarizes what happens in the `MainWindow` class
+
+To achieve this functionality, `Task` objects had to be refactored into`RealTask` and `GhostTask` objects as shown in the diagram below.
+
+![DetailedTaskClassDiagram](./images/DetailedTaskClassDiagram.png)
+
+RealTasks represent concrete tasks, which are either non-recurring tasks, or the current occurrence of recurring tasks.
+GhostTasks are temporary tasks that exist for the purpose of allowing the user to preview future occurrences of recurring tasks.
+By default, `viewTasks` will only show RealTasks.
+
+
+Since `UniqueTaskList` contains `Task` objects, it can be either `GhostTask` or `RealTask` objects. Naturally, this implies that calling two commands that both have dependencies on GhostTask objects would cause conflicts, as earlier created GhostTasks
+would still persist in the task list. This necessitates a cleanup of `GhostTask` objects between execution of each command. Such deletion of old GhostTasks in the `model` is achieved just prior to the execution of each new command in `LogicManager`, via the `deleteGhostTasks()` method.
+
+Code Snippet of the `execute(String commandText)` method in `LogicManager`:
+```
+@Override
+public CommandResult execute(String commandText) throws CommandException, ParseException {
+    logger.info("----------------[USER COMMAND][" + commandText + "]");
+    
+    //deletes all previous ghost tasks from the model as they are no longer relevant
+    model.deleteGhostTasks();
+    
+    //parsing and execution of command
+    CommandResult commandResult;
+    Command command = nurseyBookParser.parseCommand(commandText);
+    commandResult = command.execute(model);
+```
+
+
+
+#### Implementation of ViewSchedule
+
+`ViewScheduleCommand` leverages on this ability to create GhostTasks. The other unique aspect in the implementation of this feature, is how the program figures out which GhostTasks to create and show to the user upon execution of this command.
+When a `ViewScheduleCommand` is executed with a given `keyDate`, where `keyDate` refers to the date on which the user wants to view schedule, `addPossibleGhostTasksWithMatchingDate(keyDate)` is responsible for this addition of relevant GhostTasks.
+
+Each task in the task list goes through a series of checks and actions before a GhostTask is created.
+Given below is an activity diagram that summarizes the sequence of  checks and actions taken for each task in the task list upon calling the above-mentioned method.
+
+![AddPossibleGhostTasksWithMatchingDateActivityDiagram](./images/AddPossibleGhostTasksWithMatchingDateActivityDiagram.png)
+
+Since the remaining general mechanisms by which the view schedule operation occurs is similar to other previously elaborated commands, a step-by-step elaboration is not given for the overall execution.
```

Comment:
Perhaps it would be better to specify what you mean by general mechanisms? Maybe ‘how the command is parsed and how the ViewScheduleCommand is created’?

Reply:
Good suggestion! I will include this elaboration.